apps/testing-javascript/src/trpc/routers/abilities.ts

```diff
@@ -0,0 +1,30 @@
+import {getToken} from 'next-auth/jwt'
+import {
+  defineRulesForPurchases,
+  UserSchema,
+} from '@skillrecordings/skill-lesson/utils/ability'
+
+import {getSubscriberFromCookie} from '@skillrecordings/skill-lesson/utils/ck-subscriber-from-cookie'
+import {convertkitSetSubscriberCookie} from '@skillrecordings/skill-lesson/utils/ck-set-subscriber-cookie'
+import {publicProcedure, router} from '@skillrecordings/skill-lesson'
+
+export const abilities = router({
+  getAbilities: publicProcedure.query(async ({ctx, input}) => {
+    const token = await getToken({req: ctx.req})
+    const convertkitSubscriber = await getSubscriberFromCookie(ctx.req)
+
+    // TODO: where could we move this? It doesn't seem like an ability check is
+    // the best place to set a cookie.
```

Comment:
```suggestion
```

Reply:
I actually think it's a fine place to set a cookie, but the issue here is that it just sets the cookie it loads

This has the functional effect of reseting the expiration? Seems flimsy though and I didn't look to close.